### PR TITLE
*: remove `FromID` from `expression.Column`

### PR DESCRIPTION
**Pull request by winoros**
## What have you changed? (mandatory)

Remove the `FromID` from `expression.Column`.

## What is the type of the changes? (mandatory)

- Refactor

## How has this PR been tested? (mandatory)

existing test.

### CODE REVIEW

File: plan/rule_aggregation_push_down.go [hunk 1]
```diff
@@ -169,14 +169,13 @@ func (a *aggregationOptimizer) checkValidJoin(join *LogicalJoin) bool {
 
 // decompose splits an aggregate function to two parts: a final mode function and a partial mode function. Currently
 // there are no differences between partial mode and complete mode, so we can confuse them.
-func (a *aggregationOptimizer) decompose(aggFunc *aggregation.AggFuncDesc, schema *expression.Schema, id int) ([]*aggregation.AggFuncDesc, *expression.Schema) {
+func (a *aggregationOptimizer) decompose(ctx sessionctx.Context, aggFunc *aggregation.AggFuncDesc, schema *expression.Schema, id int) ([]*aggregation.AggFuncDesc, *expression.Schema) {
```
Comment (lamxTyler) [inline plan/rule_aggregation_push_down.go]: `id` is not used?

File: expression/distsql_builtin_test.go [hunk 2]
```diff
@@ -28,7 +30,38 @@ import (
...
+func (s *testEvalSuite) SetUpSuite(c *C) {
+	s.colID = 0
+}
+
+func (s *testEvalSuite) allocColID() int {
+	s.colID++
+	return s.colID
+}
+
+// generateSchema will generate a schema for test. Used only in this file.
```
Comment (lamxTyler) [inline expression/distsql_builtin_test.go]: Actually, it is only used in `schema_test.go`.

File: plan/find_best_task.go [hunk 3]
```diff
@@ -337,6 +337,14 @@ func (ds *DataSource) convertToIndexScan(prop *requiredProp, path *accessPath) (
 	}
 	rowCount := path.countAfterAccess
 	cop := &copTask{indexPlan: is}
+	names1 := make([]string, 0, len(is.Columns))
+	names2 := make([]string, 0, len(is.Index.Columns))
```
Comment (lamxTyler) [inline plan/find_best_task.go]: What's their usage?

**Comment by zz-jason** [comment]
@winoros please address the comments.

File: expression/column.go [hunk 4]
```diff
@@ -141,16 +141,13 @@ func (col *CorrelatedColumn) resolveIndices(_ *Schema) {
...
 	ColName     model.CIStr
 	DBName      model.CIStr
 	OrigTblName model.CIStr
 	TblName     model.CIStr
 	RetType     *types.FieldType
 	ID          int64
-	// Position means the position of this column that appears in the select fields.
-	// e.g. SELECT name as id , 1 - id as id , 1 + name as id, name as id from src having id = 1;
-	// There are four ids in the same schema, so you can't identify the column through the FromID and ColName.
+	// Position is the unique id of this column.
```
Comment (XuHuaiyu) [inline expression/column.go]: We may need to comment for the difference between ID and Position.

File: expression/schema.go [hunk 5]
```diff
@@ -153,13 +153,23 @@ func (s *Schema) IsUniqueKey(col *Column) bool {
...
 			return i
 		}
 	}
 	return -1
 }
 
+// FindColumnByName finds a column by its name.
+func (s *Schema) FindColumnByName(name string) *Column {
+	for _, col := range s.Columns {
+		if col.ColName.L == name {
```
Comment (XuHuaiyu) [inline expression/schema.go]: Should we take the DBName and TableName into consideration?

File: expression/column.go [hunk 6]
```diff
@@ -141,16 +141,13 @@ func (col *CorrelatedColumn) resolveIndices(_ *Schema) {
...
 	Position int
```
Comment (winoros) [inline expression/column.go]: I'll rename it to `UniqueID` or something after this pr.
I'll examine whether the `ID` of `Column` can be removed.

File: expression/column.go (hunk 4 above)
Comment (winoros) [reply expression/column.go]: This will be done is the future, not this pr.

File: expression/schema.go (hunk 5 above)
Comment (winoros) [reply expression/schema.go]: No, this is only used for datasource which only have one table.

**Comment by zz-jason** [comment]
@XuHuaiyu @lamxTyler PTAL

**Pull request merged**